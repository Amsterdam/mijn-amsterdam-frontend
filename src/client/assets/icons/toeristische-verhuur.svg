<svg width="32" height="32" viewBox="0 0 32 32" xmlns="http://www.w3.org/2000/svg" aria-label="Toeristische verhuur">
    <path d="M2.91 5v13.2h11.635V7.933h11.637C29.396 7.933 32 10.56 32 13.8V27h-2.91v-4.4H2.91V27H0V5h2.91zM8.5 8c2.49 0 4.5 2.01 4.5 4.5S10.99 17 8.5 17 4 14.99 4 12.5 6.01 8 8.5 8z" fill-rule="evenodd"/>
</svg>
