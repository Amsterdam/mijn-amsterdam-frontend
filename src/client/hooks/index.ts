export * from './analytics.hook';
export * from './detectResize.hook';
export * from './media.hook';
export * from './modalRoot.hook';
export * from './optin.hook';
export * from './pageChange';
export * from './storage.hook';
export * from './timer.hook';
export * from './useScript';
export * from './useSSE';
export * from './useThrottledFn.hook';
