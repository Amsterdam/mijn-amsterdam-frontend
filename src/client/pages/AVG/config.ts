import { AVGRequestFrontend } from '../../../server/services/avg/types';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

export const displayPropsAanvragen: DisplayProps<
  WithDetailLinkComponent<AVGRequestFrontend>
> = {
  detailLinkComponent: 'Nummer',
  ontvangstDatumFormatted: 'Ontvangen op',
  themas: 'Onderwerp(en)',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => !avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (avgVerzoek: AVGRequestFrontend) => avgVerzoek.datumAfhandeling,
    ...tableConfigBase,
  },
};

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/privacy/loket/',
    title: 'Loket persoonsgegevens gemeente Amsterdam',
  },
];
