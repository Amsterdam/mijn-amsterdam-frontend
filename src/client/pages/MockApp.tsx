import { RecoilRoot, RecoilRootProps } from 'recoil';
import React from 'react';
import { MemoryRouter, Route } from 'react-router-dom';

interface MockAppProps {
  routePath: string;
  routeEntry: string;
  initializeState?: RecoilRootProps['initializeState'];
  component: React.Component | any;
}

export default function MockApp({
  routePath,
  routeEntry,
  initializeState,
  component,
}: MockAppProps) {
  return (
    <RecoilRoot initializeState={initializeState}>
      <MemoryRouter initialEntries={[routeEntry]}>
        <Route path={routePath} component={component} />
      </MemoryRouter>
    </RecoilRoot>
  );
}
