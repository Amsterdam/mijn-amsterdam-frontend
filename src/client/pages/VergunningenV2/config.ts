export const displayPropsHuidigeVergunningen = {
  identifier: 'Kenmerk',
  title: 'Soort vergunning',
  dateRequest: 'Aangevraagd',
};

export const displayPropsEerdereVergunningen = {
  identifier: 'Kenmerk',
  title: 'Soort vergunning',
  decision: 'Resultaat',
};

export const listPageParamKind = {
  actual: 'huidige-vergunningen-en-ontheffingen',
  historic: 'eerdere-vergunningen-en-ontheffingen',
};

export type ListPageParamKey = keyof typeof listPageParamKind;
export type ListPageParamKind = (typeof listPageParamKind)[ListPageParamKey];

export const listPageTitle = {
  [listPageParamKind.actual]: 'Huidige vergunningen en ontheffingen',
  [listPageParamKind.historic]:
    'Eerdere en niet verleende vergunningen en ontheffingen',
};
