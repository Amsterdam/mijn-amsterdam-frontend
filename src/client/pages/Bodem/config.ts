import { generatePath } from 'react-router-dom';

import { LoodMetingFrontend } from '../../../server/services/bodem/types';
import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { LinkProps } from '../../../universal/types';

const displayPropsAanvragen = {
  detailLinkComponent: 'Adres',
  datumAanvraagFormatted: 'Aangevraagd',
  status: 'Status',
};

const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    ...tableConfigBase,
    title: 'Lopende aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => !bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
  },
  [listPageParamKind.completed]: {
    ...tableConfigBase,
    title: 'Afgehandelde aanvragen',
    filter: (bodemAanvraag: LoodMetingFrontend) => bodemAanvraag.processed,
    listPageRoute: generatePath(AppRoutes['BODEM/LIST'], {
      kind: listPageParamKind.completed,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    title: 'Meer informatie over lood in de bodem.',
    to: 'https://www.amsterdam.nl/wonen-leefomgeving/bodem/lood-grond/',
  },
] as const;
