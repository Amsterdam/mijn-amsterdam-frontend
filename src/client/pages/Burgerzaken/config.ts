import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { IdentiteitsbewijsFrontend, LinkProps } from '../../../universal/types';
import { withOmitDisplayPropsForSmallScreens } from '../../components/Table/helpers';
import {
  DisplayProps,
  WithDetailLinkComponent,
} from '../../components/Table/TableV2';

const displayPropsBase: DisplayProps<
  WithDetailLinkComponent<IdentiteitsbewijsFrontend>
> = {
  detailLinkComponent: 'Type',
  datumAfloopFormatted: 'Geldig tot',
};

const listPageParamKind = {
  identiteitsbewijzen: 'identiteitsbewijzen',
};

const displayProps = withOmitDisplayPropsForSmallScreens(displayPropsBase, [
  'datumAfloopFormatted',
]);

export const tableConfig = {
  [listPageParamKind.identiteitsbewijzen]: {
    title: 'Mijn reisdocumenten',
    displayProps,
    sort: dateSort('datumAfloop', 'desc'),
    listPageRoute: generatePath(AppRoutes['BURGERZAKEN/LIST'], {
      kind: listPageParamKind.identiteitsbewijzen,
    }),
  },
} as const;

export const linkListItems: LinkProps[] = [
  {
    to: 'https://www.amsterdam.nl/burgerzaken',
    title: 'Overzicht en aanvragen bij burgerzaken',
  },
] as const;

export const backLinkDetailPage = AppRoutes.BURGERZAKEN;
