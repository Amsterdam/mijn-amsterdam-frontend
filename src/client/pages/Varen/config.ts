import { generatePath } from 'react-router-dom';

import { AppRoutes } from '../../../universal/config/routes';
import { dateSort } from '../../../universal/helpers/date';
import { VarenFrontend } from '../../../server/services/varen/config-and-types';

export const displayPropsAanvragen = {
  detailLinkComponent: '',
  title: 'Omschrijving',
  dateRequestFormatted: 'Aangevraagd',
  status: 'Status',
};

export const listPageParamKind = {
  inProgress: 'lopende-aanvragen',
  completed: 'afgehandelde-aanvragen',
};

const tableConfigBase = {
  sort: dateSort('dateRequest', 'desc'),
  displayProps: displayPropsAanvragen,
};

export const tableConfig = {
  [listPageParamKind.inProgress]: {
    title: 'Lopende aanvragen',
    filter: (vergunning: VarenFrontend) => !vergunning.dateEnd,
    listPageRoute: generatePath(AppRoutes['VAREN/LIST'], {
      kind: listPageParamKind.inProgress,
    }),
    ...tableConfigBase,
  },
  [listPageParamKind.completed]: {
    title: 'Afgehandelde aanvragen',
    filter: (vergunning: VarenFrontend) => vergunning.dateEnd,
    listPageRoute: generatePath(AppRoutes['VAREN/LIST'], {
      kind: listPageParamKind.completed,
    }),
    ...tableConfigBase,
  },
};
