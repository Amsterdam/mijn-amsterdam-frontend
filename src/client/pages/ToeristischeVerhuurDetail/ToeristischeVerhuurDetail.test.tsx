import { render, screen } from '@testing-library/react';

import { generatePath } from 'react-router-dom';
import { MutableSnapshot } from 'recoil';
import vergunningenData from '../../../server/mock-data/json/vergunningen.json';
import { transformVergunningenToVerhuur } from '../../../server/services/toeristische-verhuur';
import {
  toeristischeVerhuurVergunningTypes,
  transformVergunningenData,
} from '../../../server/services/vergunningen';
import { AppRoutes } from '../../../universal/config';
import { appStateAtom } from '../../hooks/useAppState';
import MockApp from '../MockApp';
import ToeristischVerhuurDetail from './ToeristischeVerhuurDetail';

const transformedVergunningen = transformVergunningenData(
  vergunningenData as any
);

const vergunningen = transformVergunningenToVerhuur(
  transformedVergunningen.filter((vergunning: any) =>
    toeristischeVerhuurVergunningTypes.includes(vergunning.caseType)
  ) as any
);

const testState = {
  TOERISTISCHE_VERHUUR: {
    status: 'OK',
    content: { vergunningen },
  },
};

function state(state: any) {
  function initializeState(snapshot: MutableSnapshot) {
    snapshot.set(appStateAtom, state);
  }

  return initializeState;
}

describe('<ToeristischVerhuurDetail />', () => {
  jest.useFakeTimers('modern').setSystemTime(new Date('2021-06-10').getTime());

  (window as any).scrollTo = jest.fn();

  const vergunning = vergunningen?.find(
    (vergunning) =>
      vergunning.caseType === 'Vakantieverhuur' && vergunning.isActual
  );

  const routeEntry = generatePath(
    AppRoutes['TOERISTISCHE_VERHUUR/VAKANTIEVERHUUR'],
    {
      id: vergunning?.id,
    }
  );

  const routePath = AppRoutes['TOERISTISCHE_VERHUUR/VAKANTIEVERHUUR'];

  let Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={ToeristischVerhuurDetail}
      initializeState={state(testState)}
    />
  );

  it('Matches the Full Page snapshot', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Show correct properties for detail page', () => {
    render(<Component />);
    expect(screen.getAllByText('10 mei 2021').length).toBe(3);
    expect(screen.getByText('10 juli 2021')).toBeInTheDocument();
    expect(screen.getByText('14 juli 2021')).toBeInTheDocument();
    expect(screen.getByText('Ontvangen')).toBeInTheDocument();
    expect(screen.getByText('Gemeld')).toBeInTheDocument();
  });
});

describe('<ToeristischVerhuurDetail />, vergunning', () => {
  (window as any).scrollTo = jest.fn();
  const vergunning = vergunningen?.find(
    (v) => v.caseType === 'Vakantieverhuur vergunningsaanvraag'
  );
  const routeEntry = generatePath(
    AppRoutes['TOERISTISCHE_VERHUUR/VERGUNNING'],
    {
      id: vergunning?.id,
    }
  );
  const routePath = AppRoutes['TOERISTISCHE_VERHUUR/VERGUNNING'];
  let Component = () => (
    <MockApp
      routeEntry={routeEntry}
      routePath={routePath}
      component={ToeristischVerhuurDetail}
      initializeState={state(testState)}
    />
  );

  it('Show correct properties for detail page', () => {
    render(<Component />);
    expect(screen.getByText('Vergunning vakantieverhuur')).toBeInTheDocument();
    expect(screen.getByText('Vanaf')).toBeInTheDocument();
    expect(screen.getByText('Tot')).toBeInTheDocument();
    expect(screen.getByText('01 juni 2019')).toBeInTheDocument();
    expect(screen.getByText('31 mei 2020')).toBeInTheDocument();
    expect(screen.getByText('Verleend')).toBeInTheDocument();
  });
});
