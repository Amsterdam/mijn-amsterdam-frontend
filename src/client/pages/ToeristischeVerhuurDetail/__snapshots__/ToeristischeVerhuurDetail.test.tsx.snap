// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToeristischVerhuurDetail /> Matches the Full Page snapshot 1`] = `
<DocumentFragment>
  <main
    class="Page DetailPage"
  >
    <header
      class="PageHeading hasIcon hasBackLink"
    >
      <span
        aria-hidden="true"
        class="Icon"
      >
        <svg
          aria-label="Toeristische verhuur"
          class="ChapterIcon"
          fill="black"
        >
          toeristische-verhuur.svg
        </svg>
      </span>
      <div
        class="HeadingInner"
      >
        <a
          class="Button Button__plain Button__lean Linkd BackLink"
          href="/toeristische-verhuur"
        >
          <svg
            aria-hidden="true"
            class="Icon Icon__left"
            fill="#000"
            height="14px"
            width="14px"
          >
            Chevron-Left.svg
          </svg>
          Toeristische verhuur
        </a>
        <h2
          class="Heading Heading__large"
        >
          Geplande Vakantieverhuur
        </h2>
      </div>
    </header>
    <div
      class="PageContent DetailPageContent"
    >
      <div
        class="InfoDetail"
      >
        <h3
          class="Heading Heading__tiny Label"
        >
          Gemeentelijk zaaknummer
        </h3>
        <p
          class="Value"
        >
          Z/000/000005
        </p>
      </div>
      <div
        class="InfoDetail"
      >
        <h3
          class="Heading Heading__tiny Label"
        >
          Ontvangen op
        </h3>
        <p
          class="Value"
        >
          10 mei 2021
        </p>
      </div>
      <div
        class="InfoDetailGroup"
      >
        <div
          class="InfoDetailGroupContent"
        >
          <div
            class="InfoDetail"
          >
            <h3
              class="Heading Heading__tiny Label"
            >
              Datum start verhuur
            </h3>
            <p
              class="Value"
            >
              10 juli 2021
            </p>
          </div>
          <div
            class="InfoDetail"
          >
            <h3
              class="Heading Heading__tiny Label"
            >
              Datum einde verhuur
            </h3>
            <p
              class="Value"
            >
              14 juli 2021
            </p>
          </div>
        </div>
      </div>
      <div
        class="InfoDetail"
      >
        <h3
          class="Heading Heading__tiny Label"
        >
          Aantal nachten
        </h3>
        <p
          class="Value"
        >
          4
        </p>
      </div>
      <div
        class="InfoDetail LocationInfo"
      >
        <h3
          class="Heading Heading__tiny Label"
        >
          Adres
        </h3>
        <p
          class="Value"
        >
          Amstel 1 1017AB Amsterdam 
          <button
            class="Button Button__inline Button__lean MapLink"
          >
            Bekijk op de kaart
          </button>
        </p>
      </div>
    </div>
    <div
      class="StatusLine VergunningStatus"
    >
      <h4
        class="Heading Heading__tiny ListHeading"
      >
        Status
      </h4>
      <ul
        class="List"
      >
        <li
          class="ListItem"
          style="display: block;"
        >
          <div
            class="ListItemInner"
          >
            <div
              aria-label="Vinkje, processtap gereed"
              class="StatusConnection"
            >
              <span
                class="ConnectLine ConnectLine--start-checked"
              />
              <span
                class="Checkmark Checkmark--checked"
              />
            </div>
            <div
              class="Panel Panel--status"
            >
              <strong
                class="StatusTitle"
              >
                Ontvangen
              </strong>
              <time
                class="StatusDate"
              >
                10 mei 2021
              </time>
            </div>
            <div
              class="Panel Panel--description"
            >
              <div
                class="PanelContent"
              />
            </div>
            <div
              class="Panel Panel--documents"
            />
          </div>
        </li>
        <li
          class="ListItem ListItem--highlight"
          style="display: block;"
        >
          <div
            class="ListItemInner"
          >
            <div
              aria-label="Vinkje, processtap gereed"
              class="StatusConnection"
            >
              <span
                class="ConnectLine ConnectLine--end-active ConnectLine--end-checked"
              />
              <span
                class="Checkmark Checkmark--active Checkmark--checked"
              />
            </div>
            <div
              class="Panel Panel--status"
            >
              <strong
                class="StatusTitle"
              >
                Gemeld
              </strong>
              <time
                class="StatusDate"
              >
                10 mei 2021
              </time>
            </div>
            <div
              class="Panel Panel--description"
            >
              <div
                class="PanelContent"
              />
            </div>
            <div
              class="Panel Panel--documents"
            />
          </div>
        </li>
      </ul>
    </div>
  </main>
</DocumentFragment>
`;
